neuralop/models/base_model.py

```diff
@@ -76,7 +76,33 @@ def __new__(cls, *args, **kwargs):
         instance._init_kwargs = kwargs
 
         return instance
-    
+
+    def state_dict(self, **kwargs):
+        """
+        state_dict subclasses nn.Module.state_dict() and adds a metadata field
+        to track the model version and ensure only compatible saves are loaded.
+        """
+        print("In state dict")
+        state_dict = super().state_dict(**kwargs)
+        state_dict.update({'neuralop_metadata': self._init_kwargs})
+        return state_dict
+
+    def load_state_dict(self, state_dict, **kwargs):
```

Comment:
Should we match the default signature? 
https://pytorch.org/docs/stable/generated/torch.nn.Module.html

No strong preference here though since this obviates the need to update to match pytorch.